frontend/src/components/widgets/FileUploader/FileUploader.tsx

```diff
@@ -64,62 +74,105 @@ class FileUploader extends React.PureComponent<Props, State> {
     if (prevProps.disabled !== this.props.disabled && this.props.disabled) {
       this.reset()
     }
+
+    const currentMaxSize = this.props.element.get("maxUploadSizeMb")
+    if (prevProps.element.get("maxUploadSizeMb") !== currentMaxSize) {
+      this.setState({ maxSizeBytes: sizeConverter(currentMaxSize, "mb", "b") })
+    }
   }
 
-  private dropHandler = (
-    acceptedFiles: File[],
-    rejectedFiles: File[],
-    event: React.SyntheticEvent<HTMLElement>
+  public reset = (): void => {
+    this.setState({
+      status: FileStatuses.READY,
+      errorMessage: undefined,
+      files: [],
+    })
+  }
+
+  public dropHandler = (
```

Comment:
The logic contained within might change depending on product discussions. Primarily, discussions are around how to reduce reruns on file uploader changes